conn.go

```diff
@@ -198,6 +198,7 @@ func createConn(ctx context.Context, nextConn net.Conn, config *Config, isClient
 	}
 	// Do handshake
 	if err := c.handshake(ctx, hsCfg, initialFlight, initialFSMState); err != nil {
+		c.close(false)
```

Comment:
Probably want to turn it into `_ = c.close(false)`. Make it explicit that we're throwing the error away.